Improve template documentation (#180)
Concordium/concordium-rust-smart-contracts: templates/README.md

# Concordium Smart Contract Templates

> This repo helps you to get a new Concordium smart contract project up and running quickly by leveraging the pre-existing templates in this git repository. There are several smart contract templates available in this repository (currently a `default` template and a `cis2-nft` template). For generating the smart contracts from the templates, you need the `cargo-concordium` crate. `Cargo-concordium` can be installed as described in the following guide:

Reviewer: ```suggestion
This repository helps you to get a new Concordium smart contract project up and running quickly by leveraging the pre-existing templates in it. There are several smart contract templates available in this repository (currently a `default` template and a `cis2-nft` template). For generating the smart contracts from the templates, you need the `cargo-concordium` tool. `Cargo-concordium` can be installed as described in the following guide:
```

[Cargo-concordium setup](https://developer.concordium.software/en/mainnet/smart-contracts/guides/setup-tools.html#setup-tools)

> You also need the `cargo-generate` crate. `Cargo-generate` can be installed by running the following command:

Reviewer: ```suggestion
You also need the `cargo-generate` tool. `Cargo-generate` can be installed by running the following command:
```

Reviewer: Should we also mention at the start that this assumes they have installed `cargo` version at least ...?

Author: I added the following version info (cargo-generate might work with lower versions as well but I haven't tested it. I recommend the up-to-date version for cargo-generate):

- For generating the smart contracts from the templates, you need the `cargo-concordium` tool **version 2.2.0 or greater**.

- You also need the `cargo-generate` tool **(recommended version 0.16.0 or greater)**.

Reviewer: Great. looks good.

```
cargo install --locked cargo-generate
```

Here is an example of using the templates:

## Examples

### Generating a new project

> <img src="./gif_recordings/init.gif" width="700" height="450"/>

Reviewer: Could we move all of the images from the top-level into a directory like `resources`?

To start a new Concordium smart contract project from a template, run the command:

```
cargo concordium init
```

The path where the project should be created can be provided with the ``--path`` option.

### Compiling the new project

<img src="./gif_recordings/compile.gif" width="700" height="450"/>

To compile your newly generated Concordium smart contract project, run the command:

```
cargo concordium build -e --out ./my_concordium_project.wasm.v1
```

> This command will compile your new smart contract project and generate a smart contract module file named ``my_concordium_project.wasm.v1``. The ``--out`` option pastes this smart contract module file into your project root folder.

Reviewer: pastes?

Maybe outputs?

    }],
    "owner": "3uxeCZwa3SxbksPWHwXWxCsaPucZdzNaXsRbkztqUUYRo1MnvF"
}
```

### Simulating the receive function with the input parameter `throw_error == false`

<img src="./gif_recordings/simulation_receive_throw_error_false.gif" width="1000" height="350"/>

```
cargo concordium run update --entrypoint receive --module ./my_concordium_project.wasm.v1 --parameter-json throwErrorFalse.json --state-bin state.bin --contract my_concordium_project --context contextUpdate.json
```

This command simulates an invoke to the ``receive`` function of your smart contract module with the input parameter `throw_error == false`.

### Simulating the receive function with the input parameter `throw_error == true`

<img src="./gif_recordings/simulation_receive_throw_error_true.gif" width="1000" height="350"/>

```
cargo concordium run update --entrypoint receive --module ./my_concordium_project.wasm.v1 --parameter-json throwErrorTrue.json --state-bin state.bin --contract my_concordium_project --context contextUpdate.json
```

This command simulates an invoke to the ``receive`` function of your smart contract module with the input parameter `throw_error == true`.

## The `cis2-nft` Template

> This template generates an NFT token smart contract. The token follows the Concordium [CIS2 token standard](https://proposals.concordium.software/CIS/cis-2.html). To get better familiar with how to interact with a CIS2 token, you can read the

Reviewer: ```suggestion
This template generates an NFT token smart contract. The token follows the Concordium [CIS2 token standard](https://proposals.concordium.software/CIS/cis-2.html). To get more familiar with how to interact with a CIS2 token, you can read the
```
